src/hist/core.py

```diff
@@ -539,6 +539,8 @@ def plot_pull(
         fig.add_axes(ax)
         fig.add_axes(pull_ax)
 
+        plt.close(fig)
```

Comment:
You have to close in the test, not in the source - otherwise, no one will ever be able to use the figure!

Reply:
yep, I did this because the plt.close did not work in test files so I came up with this idea. Okay, I will TAL at mpl to see how it works, too.

Reply:
No, you are right. This should have fixed the problem, even though it's not a final solution. Sorry! Will have to play with ideas here, then.

Reply:
Though you could try adding this to all ~4 functions, just to make sure it's not the other 3 that are overloading the open plots count.

Reply:
> Though you could try adding this to all ~4 functions, just to make sure it's not the other 3 that are overloading the open plots count.

One warning lose. Good starting point. But still 35 warnings remained.